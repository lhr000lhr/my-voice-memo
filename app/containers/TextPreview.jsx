import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import { View, StyleSheet, ActivityIndicator } from 'react-native'
import { Text } from 'react-native-elements'
import { createAction } from '../utils'

function TextPreview({ dispatch, route, textPreview }) {
  useEffect(() => {
    dispatch(createAction('textPreview/queryText')({ recording: route.params.recording }))
  }, [])

  const { loading, recording } = textPreview
  return (
    <View style={styles.container}>
      {loading ? (
        <ActivityIndicator animating={loading} />
      ) : (
        <Text>{recording?.content?.DisplayText}</Text>
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  listStyle: {
    width: '100%',
  },
  buttonStyle: {
    height: '100%',
    width: 100,
    alignSelf: 'flex-end',
    backgroundColor: 'red',
  },
})

export default connect(({ textPreview }) => ({ textPreview }))(TextPreview)
