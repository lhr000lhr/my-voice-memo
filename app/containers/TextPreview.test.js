import React from 'react'
import renderer from 'react-test-renderer'
import TextPreview from './TextPreview'

jest.mock('react-redux', () => ({
  connect: () => Component => Component,
}))

const props = {
  textPreview: {
    recording: {},
    loading: false,
  },
}

describe('TextPreview', () => {
  it('renders correctly with textPreview', () => {
    expect(renderer.create(<TextPreview {...props} />)).toMatchSnapshot()
  })
})
