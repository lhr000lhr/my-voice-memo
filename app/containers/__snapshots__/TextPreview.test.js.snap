// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextPreview renders correctly with textPreview 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#242424",
      }
    }
  >
    
  </Text>
</View>
`;

exports[`TextPreview renders correctly with textPreview screen empty 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#242424",
      }
    }
  >
    
  </Text>
</View>
`;

exports[`TextPreview renders correctly with textPreview should be loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#242424",
      }
    }
  >
    
  </Text>
</View>
`;
